src/sage/rings/polynomial/multi_polynomial_ideal.py

```diff
@@ -2854,10 +2869,14 @@ def _variety(T, V, v=None):
         P = self.ring()
         if ring is not None:
             P = P.change_ring(ring)
-        try:
-            TI = self.triangular_decomposition('singular:triangLfak')
-            T = [list(each.gens()) for each in TI]
-        except TypeError:  # conversion to Singular not supported
+        T = None
```

Comment:
So this is basically the same code as before but you only try if the characteristic is small enough? Looks good.

Reply:
yes, I find it cleanest to implement the logic with a `T = None` beforehand.